packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/features/entityFeatures.ts

```diff
@@ -191,6 +196,216 @@ function cacheGetNeighborEntityElement(
+                          : (position.node.parentNode as HTMLElement).className ===
+                            DelimiterClasses.DELIMITER_BEFORE
+                          ? position.node.parentElement
+                          : null;
+                  } else if (position.node == position.element) {
+                      return position.element.childNodes.item(position.offset);
+                  } else {
+                      return null;
+                  }
+              },
+              isAtEndOrBeginning: position.isAtEnd,
+          }
+        : {
+              class: DelimiterClasses.DELIMITER_AFTER,
+              pairClass: DelimiterClasses.DELIMITER_BEFORE,
+              getDelimiterPair: (element: HTMLElement) =>
+                  element.previousElementSibling?.previousElementSibling,
+              getNextSibling: () => {
+                  if (position.node.nodeType == NodeType.Text) {
+                      return position.node.previousSibling != null
+                          ? position.node.previousSibling
+                          : (position.node.parentNode as HTMLElement).className ===
+                            DelimiterClasses.DELIMITER_AFTER
+                          ? position.node.parentElement
+                          : null;
+                  } else if (position.node == position.element) {
+                      return position.element.childNodes.item(position.offset);
+                  } else {
+                      return null;
+                  }
+              },
+              isAtEndOrBeginning: position.offset == 0,
+          };
+
+    position.element.normalize();
+    if (data.isAtEndOrBeginning && data.getNextSibling()) {
+        const elAtCursor = editor.getElementAtCursor('.' + data.class, data.getNextSibling());
+
+        if (!!shouldHandle(elAtCursor)) {
+            return true;
+        }
+    }
+
+    const entityAtCursor = editor.getElementAtCursor('.' + data.class, focusedElement);
+
+    return !!shouldHandle(entityAtCursor);
+
+    function shouldHandle(element: HTMLElement) {
+        return (
+            element &&
+            data.getDelimiterPair(element)?.className.indexOf(data.pairClass) > -1 &&
+            cacheDelimiter(event, checkBefore, element)
+        );
+    }
+}
+
+function cacheDelimiter(event: PluginEvent, checkBefore: boolean, delimiter?: HTMLElement | null) {
+    return cacheGetEventData(event, 'delimiter_cache_key_' + checkBefore, () => delimiter);
+}
+
+function cacheEntityBetweenDelimiter(
+    event: PluginKeyboardEvent,
+    editor: IEditor,
+    checkBefore: boolean,
+    entity?: Element | null,
+    operation?: EntityOperation
+) {
+    const element = cacheGetEventData(event, 'entity_delimiter_cache_key_' + checkBefore, () =>
+        editor.getElementAtCursor(getEntitySelector(), entity)
+    );
+
+    if (element && operation !== undefined) {
+        editor.triggerPluginEvent(PluginEventType.EntityOperation, {
```

Comment:
Do we expect plugin to help delete the delimiters?

Reply:
I added some logic in previous PR to delete the delimeters when a entity is deleted [here](https://github.com/microsoft/roosterjs/blob/0341fa88ecb018f2f09964ee87249f6848c94a31/packages/roosterjs-editor-core/lib/corePlugins/utils/InlineEntityHandlers/inlineEntityOnPluginEvent.ts#L41). Not sure if that is okay or we should move the logic elsewhere